### PR TITLE
chore(resource): check if the file exists before uploading

### PR DESCRIPTION
**Pull request by hunshenshi**
https://app.clickup.com/t/866ae8gbe

### CODE REVIEW

File: pkg/depends/conf/filesystem/amazonS3/s3_config.go [hunk 1]
```diff
@@ -95,3 +98,28 @@ func (s *AmazonS3) DownloadUrl(key string) (string, error) {
...
+		Bucket: aws.String(s.BucketName),
+		Key:    aws.String(key),
+	})
+	if err != nil {
+		awsErr, ok := err.(awserr.RequestFailure)
+		if ok && awsErr.StatusCode() == 404 {
+			return nil, filesystem.ErrNotExistObjectKey
+		} else {
+			return nil, err
+		}
```
Comment (CoderZhi) [inline pkg/depends/conf/filesystem/amazonS3/s3_config.go:113]:
```
    }
    return nil, err
```

File: pkg/depends/conf/filesystem/s3/s3.go [hunk 2]
```diff
@@ -175,24 +174,23 @@ func (db *ObjectDB) DeleteObject(ctx context.Context, meta *ObjectMeta) error {
...
-	defer object.Close()
-
-	info, err := object.Stat()
-	if err != nil {
-		return nil, err
+		awsErr, ok := err.(awserr.RequestFailure)
+		if ok && awsErr.StatusCode() == 404 {
+			return nil, filesystem.ErrNotExistObjectKey
+		} else {
+			return nil, err
```
Comment (CoderZhi) [inline pkg/depends/conf/filesystem/s3/s3.go:189]: same here

File: pkg/depends/conf/filesystem/s3/s3.go [hunk 3]
```diff
@@ -274,23 +272,8 @@ func (db *ObjectDB) DownloadUrl(key string) (string, error) {
...
-	oid, err := strconv.ParseUint(parts[1], 10, 64)
-	if err != nil {
-		return nil, ErrInvalidObjectKey
-	}
-
-	om := &ObjectMeta{ObjectID: oid, Group: grp}
-
-	return om, nil
+func (db *ObjectDB) StatObject(key string) (*filesystem.ObjectMeta, error) {
+	return nil, nil
```
Comment (CoderZhi) [inline pkg/depends/conf/filesystem/s3/s3.go:276]: why return nil for s3?

Comment (hunshenshi) [reply pkg/depends/conf/filesystem/s3/s3.go:276]: s3 will remove, just use amazonS3 pkg. I will add Deprecated.

File: pkg/modules/resource/resource.go [hunk 4]
```diff
@@ -41,8 +41,15 @@ func Create(ctx context.Context, acc types.SFID, fh *multipart.FileHeader, filen
 		},
 		func(d sqlx.DBExecutor) error {
 			if found {
+				if !CheckExist(ctx, res.ResourceID) {
+					_, err := UploadFile(ctx, data, res.ResourceID)
+					if err != nil {
+						return err
```
Comment (Liuhaai) [inline pkg/modules/resource/resource.go:47]: could the system work without `s3`?

Comment (hunshenshi) [reply pkg/modules/resource/resource.go:47]: yes, it will use local file system without s3.

**Review by CoderZhi — APPROVED** [review]

**Pull request merged**